### PR TITLE
Move the source of truth of kaggle-api development to this repository

### PR DESCRIPTION
**Pull request by Philmod**
## Changes

- It uses `hatch`, for example `hatch run compile` to generate the python files to release to pypi
- The generated code is checked-in so that users can pip install from source, as before this change
- The important files to review are: 
  - the new README.md
  - the tools/GeneratePythonLibrary.sh that was modified

## Next

- Remove code from main kaggle repo
- Internal documentation for release, etc...

### Ideas for medium-term

- Lint / Coverage (example: https://github.com/Kaggle/kagglehub?tab=readme-ov-file#lint--format)
- Unit tests (run on PR)
- Integration tests (run on PR)
- Script to change the version across the 4 files

## Bug

http://b/325046928

### CODE REVIEW

**Review by erdalsivri — APPROVED** [review]
LGTM!

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,946 @@
...
+
+```
+usage: kaggle models get [-h] [-p FOLDER] [model]
+
+required arguments:
+  model                 Model URL suffix in format <owner>/<model-name>
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -p PATH, --path PATH  Folder where the special model-metadata.json file (https://github.com/Kaggle/kaggle-api/wiki/Model-Metadata) will be downloaded (if specified).
```
Comment (stevemessick) [inline docs/README.md:587]: Line break.

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,946 @@
...
+```
+
+Example:
+
+`kaggle models get tensorflow/toxicity`
+
+##### List models
+
+```
+usage: kaggle models list [--sort-by SORT_BY] [-s SEARCH] [--owner OWNER] [--page-token PAGE_TOKEN] [--page-size PAGE_SIZE] [--csv]
```
Comment (stevemessick) [inline docs/README.md:597]: Line break.

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,946 @@
...
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --sort-by SORT_BY     Sort list results. Default is 'hotness'. Valid options are 'hotness', 'downloadCount', 'voteCount', 'notebookCount' and 'createTime'
```
Comment (stevemessick) [inline docs/README.md:601]: Line break.

File: docs/README.md [hunk 4]
```diff
@@ -0,0 +1,946 @@
...
+
+##### Initialize metadata file for a model
+
+```
+usage: kaggle models init [-h] [-p FOLDER]
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -p FOLDER, --path FOLDER
+                        Folder to create the model-metadata.json file (https://github.com/Kaggle/kaggle-api/wiki/Model-Metadata). Defaults to current working directory
```
Comment (stevemessick) [inline docs/README.md:625]: Line break.

File: docs/README.md [hunk 5]
```diff
@@ -0,0 +1,946 @@
...
+
+If you want to create a new model, you need to initiate metadata file at first. You could fulfill this by running `kaggle models init` as describe above.
+
+```
+usage: kaggle models create [-h] [-p FOLDER]
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -p FOLDER, --path FOLDER
+                        Folder containing the special model-metadata.json file (https://github.com/Kaggle/kaggle-api/wiki/Model-Metadata). Defaults to current working directory
```
Comment (stevemessick) [inline docs/README.md:642]: Line break.

File: docs/README.md [hunk 6]
```diff
@@ -0,0 +1,946 @@
...
+
+If you want to update a model, you need a metadata file at first. You can fetch the data by running `kaggle models get owner/slug -p folder`.
+
+```
+usage: kaggle models update [-h] [-p FOLDER]
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -p FOLDER, --path FOLDER
+                        Folder containing the special model-metadata.json file (https://github.com/Kaggle/kaggle-api/wiki/Model-Metadata). Defaults to current working directory
```
Comment (stevemessick) [inline docs/README.md:675]: Line break.

File: docs/README.md [hunk 7]
```diff
@@ -0,0 +1,946 @@
...
+
+```
+usage: kaggle models instances get [-h] [-p FOLDER] [modelInstance]
+
+required arguments:
+  modelInstance         Model Instance URL suffix in format <owner>/<model-name>/<framework>/<instance-slug>
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -p PATH, --path PATH  Folder where the special model-instance-metadata.json file (https://github.com/Kaggle/kaggle-api/wiki/Model-Metadata) will be downloaded (if specified).
```
Comment (stevemessick) [inline docs/README.md:712]: Line break.

File: docs/README.md [hunk 8]
```diff
@@ -0,0 +1,946 @@
...
+
+##### Initialize metadata file for a model instance
+
+```
+usage: kaggle models instances init [-h] [-p FOLDER]
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -p FOLDER, --path FOLDER
+                        Folder to create the model-instance-metadata.json file (https://github.com/Kaggle/kaggle-api/wiki/Model-Metadata). Defaults to current working directory
```
Comment (stevemessick) [inline docs/README.md:727]: Line break.

File: docs/README.md [hunk 9]
```diff
@@ -0,0 +1,946 @@
...
+```
+
+Example:
+
+`kaggle models instances init -p /path/to/modelinstance`
+
+##### Create a new model instance
+
+If you want to create a new model instance, you need to initiate metadata file at first. You could fulfill this by running `kaggle models instances init` as describe above.
```
Comment (stevemessick) [inline docs/README.md:736]: Similar to before, I think "initiate" should be "initialize the" and "fulfill" should be "do". Also "describeD".

File: docs/README.md [hunk 10]
```diff
@@ -0,0 +1,946 @@
...
+
+```
+usage: kaggle models instances create [-h] [-p FOLDER]
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -p FOLDER, --path FOLDER
+                        Folder containing the special model-instance-metadata.json file (https://github.com/Kaggle/kaggle-api/wiki/Model-Metadata). Defaults to current working directory
```
Comment (stevemessick) [inline docs/README.md:744]: Line break.

File: docs/README.md [hunk 11]
```diff
@@ -0,0 +1,946 @@
...
+  -q, --quiet           Suppress printing information about the upload progress
+  -r {skip,zip,tar}, --dir-mode {skip,zip,tar}
+                        What to do with directories: "skip" - ignore; "zip" - compressed upload; "tar" - uncompressed upload
```
Comment (stevemessick) [inline docs/README.md:747]: Line break.

File: docs/README.md [hunk 12]
```diff
@@ -0,0 +1,946 @@
...
+  -h, --help     show this help message and exit
+```
+
+Example:
+
+`kaggle models instances delete tensorflow/toxicity/tfjs/default`
+
+##### Update a model instance
+
+If you want to update a model instance, you need a metadata file at first. You can fetch the data by running `kaggle models instances get owner-slug/model-slug/framework/instance-slug -p folder`.
```
Comment (stevemessick) [inline docs/README.md:772]: Line break.

File: docs/README.md [hunk 13]
```diff
@@ -0,0 +1,946 @@
...
+
+```
+usage: kaggle models instances update [-h] [-p FOLDER]
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -p FOLDER, --path FOLDER
+                        Folder containing the special model-instance-metadata.json file (https://github.com/Kaggle/kaggle-api/wiki/Model-Metadata). Defaults to current working directory
```
Comment (stevemessick) [inline docs/README.md:780]: Line break.

File: docs/README.md [hunk 14]
```diff
@@ -0,0 +1,946 @@
...
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -p FOLDER, --path FOLDER
+                        Folder containing the model files to upload
+  -n, --version-notes NOTES
+                        Version notes to record for this new version
+  -q, --quiet           Suppress printing information about the upload progress
+  -r {skip,zip,tar}, --dir-mode {skip,zip,tar}
+                        What to do with directories: "skip" - ignore; "zip" - compressed upload; "tar" - uncompressed upload
```
Comment (stevemessick) [inline docs/README.md:821]: Line break.

File: docs/README.md [hunk 15]
```diff
@@ -0,0 +1,946 @@
...
+
+`kaggle models instances versions create tensorflow/toxicity/tfjs/default -p /path/to/files -n "updated weights"`
+
+##### Download a model instance version
+
+```
+usage: kaggle models instances versions download [-h] [-p PATH] [--untar] [-f] [-q] [modelInstanceVersion]
+
+required arguments:
+  modelInstanceVersion  Model Instance version URL suffix in format <owner>/<model-name>/<framework>/<instance-slug>/<version_number>
```
Comment (stevemessick) [inline docs/README.md:834]: Line break.

File: docs/README.md [hunk 16]
```diff
@@ -0,0 +1,946 @@
...
+
+`kaggle models instances versions download tensorflow/toxicity/tfjs/default/1`
+
+##### Delete model instance
+
+```
+usage: kaggle models instances versions delete [-h] [modelInstanceVersion]
+
+required arguments:
+  modelInstanceVersion  Model Instance version URL suffix in format <owner>/<model-name>/<framework>/<instance-slug>/<version_number>
```
Comment (stevemessick) [inline docs/README.md:855]: Line break.

**Review by stevemessick — APPROVED** [review]
I made lots of comments about line breaks. Those can be considered optional. I did clean up the language in a few places and those should be included.

**Comment by stevemessick** [comment]
Also, note that when you first see my comments you may have a bunch that are collapsed. I had to click the line that said there were 26 hidden comments to see them all.

**Review by neshdev — APPROVED** [review]
LGTM. 

I would have assume that the delta between the repos to be a lot smaller. Has the repo not been synced for a while?

Also, is it possible to get ride of the formatting changes?

**Pull request merged**